app/src/main/java/com/example/nostack/views/StartUp.java

```diff
@@ -157,6 +158,11 @@ public void onClick(View v) {
         view.findViewById(R.id.AdministratorSignInButton).setOnClickListener(new View.OnClickListener(){
             @Override
             public void onClick(View v){
+                boolean isAdmin = profile.getRole() != null && profile.getRole().contains("admin");
+                if(!isAdmin){
+                    Toast.makeText(getActivity(), "Unauthorized", Toast.LENGTH_SHORT).show();
```

Comment:
Should we make this message have slightly more context? Such as "Unauthroized to access this information" or something

Reply:
i think it works,